pkg/component/kubernetes/apiserver/apiserver_test.go

```diff
@@ -3177,6 +3180,7 @@ kind: AuthorizationConfiguration
 						Protocol:      corev1.ProtocolTCP,
 					}))
 					Expect(deployment.Spec.Template.Spec.Containers[0].Resources).To(Equal(apiServerResources))
+					Expect(deployment.Spec.Template.Spec.Containers[0].SecurityContext).To(Equal(apiServerSecurityContext))
```

Comment:
I don't think we need the `apiServerSecurityContext` var. Maybe you copy-pasted from the above assertion for the resources and created such var similar to the `apiServerResources` var. However, `apiServerResources` is being used in several place - it is being passed as value of the component.

```suggestion
					Expect(deployment.Spec.Template.Spec.Containers[0].SecurityContext).To(Equal(&corev1.SecurityContext{
						AllowPrivilegeEscalation: ptr.To(false),
					}))
```